Adding docs for backup, TLS and status fields docs (#28)
appuio/appcat-docs: docs/modules/ROOT/pages/vshn-managed/postgresql/usage.adoc

[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:

>   name: widera-testing

Reviewer: Please use an anonymous name for examples.

  labels:
    name: widera-testing
---
apiVersion: vshn.appcat.vshn.io/v1
kind: VSHNPostgreSQL
metadata:
  name: my-postgres-example
  namespace: widera-testing
spec:
  parameters:
    service:
      majorVersion: "15"
    backup:
      schedule: '0 22 * * *'
  writeConnectionSecretToRef:
    name: postgres-creds-connection
---
apiVersion: v1
kind: Pod
metadata:
  name: postgres-client
  namespace: widera-testing
spec:
  containers:
  - name: postgres
    image: postgres:15
    resources:
      limits:
        memory: "128Mi"
        cpu: "500m"
    command:
      - /bin/sh
      - -c
    args:
      - PGPASSWORD=$POSTGRESQL_PASSWORD psql "sslmode=verify-ca sslrootcert=/etc/secret-volume/ca.crt host=$POSTGRESQL_HOST port=$POSTGRESQL_PORT dbname=$POSTGRESQL_DB" -U $POSTGRESQL_USER -l
    envFrom:
      - secretRef:
          name: postgres-creds-connection
    volumeMounts:
    - name: secret-volume
      readOnly: true
      mountPath: "/etc/secret-volume"
  volumes:
  - name: secret-volume
    secret:
      defaultMode: 0600
      secretName: postgres-creds-connection
  restartPolicy: OnFailure
----

> == Debug instance by Yourself

Reviewer: ```suggestion
== Debug instance by yourself
```

> You can easily check what happend during creation by checking status fields. PostgreSQL managed by VSHN is composed from many resources, responsible for

Reviewer: ```suggestion
You can easily check what happened during creation by checking the status fields. PostgreSQL by VSHN is composed from many resources, responsible for
```

Please do not insert a newline in the middle of a sentence. AsciiDoc is meant to be readable in the source as well.

cluster creation, bucket creation, backups and uploads to cloud, each of those components might break or be misconfigured (f.e typo). You can check what happend by running

```

> # example assumes reuse of names and namespace from example above

Reviewer: Then please ensure it really correlates :smile:

> k -n development describe vshnpostgresqls.vshn.appcat.vshn.io my-postgres-example 

Reviewer: ```suggestion
kubectl -n development describe vshnpostgresqls.vshn.appcat.vshn.io my-postgres-example 
```

Don't assume the user has an alias `k`

```

Example output looks as follows:

```
Status:

>   Certificate Debug:

Reviewer: Why is this called "... Debug"? IMHO that's a very strange naming for a status field, I've never seen that in any Kubernetes resource this way. "Debug" implies to me that I increased the log level somewhere and now it tells my messages from a debugging log level, which I would not want to see in production.

Author: I called that `...Debug` because purpose of that field is to help our customers debug their instances

Reviewer: I must admit: I don't like it at all. Up to a point where I'd like to ask to change that. My proposal would be to call that "... Conditions". Because as it looks like it's a copy of the "Conditions" field from other resources, is it?

Author: No problem, I'll create a fastlane ticket for that

    Type:                  Ready
    Last Transition Time:  2023-02-13T15:11:43Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
  Pgcluster Debug:
    Last Transition Time:  2023-02-13T15:11:48Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2023-02-13T15:11:47Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
  Pgconfig Debug:
    Last Transition Time:  2023-02-13T15:11:44Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2023-02-13T15:11:43Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
  Profile Debug:
    Last Transition Time:  2023-02-13T15:11:44Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2023-02-13T15:11:42Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced

>   s3BackupConfigDebug:

Reviewer: Is this by intention that this is named differently? All other status fields with "Debug" in the name do have an uppercase first letter and a space before "Debug".

Author: this is probably just formatting from `kubectl describe` it doesn't want to split words when it see number in name

    Last Transition Time:  2023-02-13T15:11:48Z
    Reason:                Available
    Status:                True
    Type:                  Ready
    Last Transition Time:  2023-02-13T15:11:46Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
  s3BucketDebug:
    Last Transition Time:  2023-02-13T15:11:42Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
    Last Transition Time:  2023-02-13T15:11:47Z
    Reason:                Available
    Status:                True
    Type:                  Ready
  Secret Debug:
    Last Transition Time:  2023-02-13T15:12:07Z
    Reason:                ReconcileSuccess
    Status:                True
    Type:                  Synced
    Last Transition Time:  2023-02-13T15:12:13Z
    Reason:                Available
    Status:                True
    Type:                  Ready
```

> If there will be any issues with Your instance, they will be printed in above status fields

Reviewer: ```suggestion
If there are any issues with your instance, they will be mentioned in the status fields.
```
